### PR TITLE
Update Spin CLI to 1.24.0 for Mercari release

### PR DESCRIPTION
**Pull request by KeisukeYamashita**
## WHAT
As title.

## WHY
What's to use the latest Spinnaker version's Spin CLI.

### CODE REVIEW

**Review by dragon3 — COMMENTED** [review]
The change itself looks good to me 👍 
Suggestion and question:

- `make fmtcheck` fails with the following message, it's better to fix it.
  ```
  ➜  terraform-provider-spinnaker git:(v0.3.0-mercari-update/spin-cli) make fmtcheck
  ==> Checking that code complies with gofmt requirements...
  gofmt needs running on the following files:
  ./spinnaker/resource_application_test.go
  You can use the command: `make fmt` to reformat code.
  make: *** [fmtcheck] Error 1
  ```
- There are some changes between `master` and `v0.3.0-mercari`. Is it better to merge them as much as possible?

**Comment by KeisukeYamashita** [comment]
@dragon3  I ported some of the commits in the master & I ran the test again and works well 👍

File: .github/workflows/release-provider.yml [hunk 1]
```diff
@@ -13,10 +13,15 @@ jobs:
...
-        uses: actions/setup-go@v1
+        uses: actions/setup-go@v2.1.3
         with:
+<<<<<<< HEAD
           go-version: 1.13
       - uses: actions/cache@v1
+=======
+          go-version: 1.15
+      - uses: actions/cache@v2.1.3
+>>>>>>> 813e646... Merge pull request #92 from mercari/upgrade-go-version
```
Comment (dragon3) [inline .github/workflows/release-provider.yml:24]:
```suggestion
          go-version: 1.15
      - uses: actions/cache@v2.1.3
```

**Review by dragon3 — COMMENTED** [review]
Thanks!

I found conflict remaining in `.github/workflows/release-provider.yml` , please fix it.

And it seems that the following changes are not ported, is it ok?
- https://github.com/mercari/terraform-provider-spinnaker/pull/77
- https://github.com/mercari/terraform-provider-spinnaker/pull/80
- https://github.com/mercari/terraform-provider-spinnaker/pull/88

File: .github/workflows/release-provider.yml (hunk 1 above)
Comment (KeisukeYamashita) [reply .github/workflows/release-provider.yml:24]: Oh, I missed that 😢  I will fix right away 🏃  Thank you 🙇

Comment (KeisukeYamashita) [reply .github/workflows/release-provider.yml]: I edited the commit of `Merge pull request #92 from mercari/upgrade-go-version` which contained that unresolved merge conflicts. And I ported(cherry-picked) the three PRs you suggested. And then I force pushed. 

PTAL 🙇

**Review by dragon3 — APPROVED** [review]
LGTM 👍 
( There still are some diffs related to goreleaser, but it should be fine for now ).

**Comment by KeisukeYamashita** [comment]
> ( There still are some diffs related to goreleaser, but it should be fine for now ).

For this, I created a new PR 🙇 
https://github.com/mercari/terraform-provider-spinnaker/pull/100

**Pull request merged**